aml/org.testeditor.aml.dsl/src/org/testeditor/aml/dsl/scoping/MethodReferenceScopes.xtend

```diff
@@ -32,37 +35,41 @@ class MethodReferenceScopes {
 	@Inject extension IQualifiedNameConverter 
 	
 	/**
-	 * Delegate to appropiate methods depending on the passed reference.
+	 * Provides scoping for MethodReference.operation.
 	 */
-	def IScope getMethodReferenceScope(MethodReference context, EReference reference) {
-		return switch (reference) {
-			case METHOD_REFERENCE__OPERATION: context.scopeFor_MethodReference_operation 
-			// TODO provide scope for type as well - maybe use IImportsConfiguration
-			default: IScope.NULLSCOPE
-		}
+	def IScope scopeFor_MethodReference_operation(MethodReference context) {
+		val fixtureMethods = getFixtureMethods(context.typeReference)
+		return Scopes.scopeFor(fixtureMethods, [simpleName.toQualifiedName], IScope.NULLSCOPE)
 	}
-	
+
 	/**
-	 * Provides scoping for MethodReference.operation.
+	 * Provides scoping for MethodReference.parameters.
 	 */
-	protected def IScope scopeFor_MethodReference_operation(MethodReference context) {
-		val type = context.typeReference?.type
-		if (type instanceof JvmDeclaredType) {
-			val fixtureMethods = type.fixtureMethods
-			return Scopes.scopeFor(fixtureMethods, [simpleName.toQualifiedName], IScope.NULLSCOPE)
-		}
-		return IScope.NULLSCOPE	
+	def IScope scopeFor_MethodReference_parameters(MethodReference context) {
+		val interactionType = context.getContainerOfType(InteractionType)
+		val availableVariables = interactionType.template.contents.filter(TemplateVariable)
+		return Scopes.scopeFor(availableVariables)
 	}
-	
+
 	/**
-	 * @return all public, non-static operations that are annotated with @FixtureMethod, inherited methods included.
+	 * @param typeReference may be {@code null}
+	 * @return all public, non-static operations that are annotated with @FixtureMethod, inherited methods included, never {@code null}
 	 */
+	def List<JvmOperation> getFixtureMethods(JvmParameterizedTypeReference typeReference) {
+		val result = newLinkedList
```

Comment:
Do you really need a modifiable List to be returned here? Why not return getFixtureMethods(type) itself and an empty list in the else case?